### PR TITLE
relayer: improve delivery process debugging, add utility functions

### PR DESCRIPTION
**Pull request by chase-45**

### CODE REVIEW

File: relayer/generic_relayer/pricing/src/utils/currentPricing.ts
Comment (derpy-duck) [inline relayer/generic_relayer/pricing/src/utils/currentPricing.ts:117]: Is this a xLabs specific pricing process? or a generic one
if the latter, I don't think we can expect that the delivery provider exposes (maximumBudget, nativeCurrencyPrice, assetConversionBuffer, etc) right

File: Tiltfile
Comment (JoeHowarth) [inline Tiltfile:488]: nit: wormhole relayer instead of generic relayer

File: relayer/generic_relayer/pricing/src/app.ts
Comment (JoeHowarth) [inline relayer/generic_relayer/pricing/src/app.ts:53]: How is this suppossed to be triggered? It appears to be listening for deliveryVaas and running the pricing based off that? What's the rationale here vs. a background job?

**Review by JoeHowarth — COMMENTED** [review]
Can you write a readme describing how the pricing piece should be used?

Can we also get a review from @solanoepalacio? My understanding was someone on his team was also working on the pricing oracle updater

File: relayer/generic_relayer/pricing/src/utils/currentPricing.ts
Comment (chase-45) [reply relayer/generic_relayer/pricing/src/utils/currentPricing.ts:117]: This is meant for anyone who intends to use the reference DeliveryProvider contracts

File: relayer/generic_relayer/pricing/src/app.ts
Comment (chase-45) [reply relayer/generic_relayer/pricing/src/app.ts:53]: It would be good in the future to trigger based off a cron timer or something similar, but currently the trigger isn't meaningful, I agree

**Comment by derpy-duck** [comment]
Probably review shouldn't come from me (maybe Joe or Solano?) but just quick note that it would be nice if after you merge this to generic-relayer-merge, to cherry-pick that commit into #3042

**Review by JoeHowarth — DISMISSED** [review]

**Review by derpy-duck — APPROVED** [review]

**Pull request merged**
